example/java/TestBufferStreamVCF2TileDB.java

```diff
@@ -0,0 +1,169 @@
+    private static class LinkedHashFactory implements ContainerFactory
+    {
+        public List creatArrayContainer()
+        {
+            return new ArrayList();
+        }
+
+        public Map createObjectContainer()
+        {
+            return new LinkedHashMap();
+        }
+    }
+
+    public static void main(final String[] args) throws IOException, FileNotFoundException, GenomicsDBException, ParseException
+    {
+        if(args.length < 2)
+        {
+            System.err.println("For loading: [-iterators] <loader.json> <stream_name_to_file.json> [bufferCapacity rank lbRowIdx ubRowIdx]");
+            System.exit(-1);
+        }
+        int argsLoaderFileIdx = 0;
+        if(args[0].equals("-iterators") || args[0].equals("-mv_iterator"))
+            argsLoaderFileIdx = 1;
+        //Buffer capacity
+        long bufferCapacity = (args.length >= argsLoaderFileIdx+3) ? Integer.parseInt(args[argsLoaderFileIdx+2]) : 1024;
+        //Specify rank (or partition idx) of this process
+        int rank = (args.length >= argsLoaderFileIdx+4) ? Integer.parseInt(args[argsLoaderFileIdx+3]) : 0;
+        //Specify smallest row idx from which to start loading - useful for incremental loading into existing array
+        long lbRowIdx = (args.length >= argsLoaderFileIdx+5) ? Long.parseLong(args[argsLoaderFileIdx+4]) : 0;
+        //Specify largest row idx up to which loading should be performed - for completeness
+        long ubRowIdx = (args.length >= argsLoaderFileIdx+6) ? Long.parseLong(args[argsLoaderFileIdx+5]) : Long.MAX_VALUE-1;
+        //<loader.json> first arg
+        VCF2TileDB loader = new VCF2TileDB(args[argsLoaderFileIdx], rank, lbRowIdx, ubRowIdx);
+        //<stream_name_to_file.json> - useful for the driver only
+        //JSON file that contains "stream_name": "vcf_file_path" entries
+        FileReader mappingReader = new FileReader(args[argsLoaderFileIdx+1]);
+        JSONParser parser = new JSONParser();
+        LinkedHashMap streamNameToFileName = (LinkedHashMap)parser.parse(mappingReader, new LinkedHashFactory());
+        ArrayList<VCFFileStreamInfo> streamInfoVec = new ArrayList<VCFFileStreamInfo>();
+        long rowIdx = 0;
+        for(Object currObj : streamNameToFileName.entrySet())
+        {
+            Map.Entry<String, String> entry = (Map.Entry<String, String>)currObj;
+            VCFFileStreamInfo currInfo = new VCFFileStreamInfo(entry.getValue());
+            //The following 2 lines are not mandatory - use initializeSampleInfoMapFromHeader() iff you know for sure that sample names
+            //in the VCF header are globally unique across all streams/files. If not, you have 2 options:
+            //(a) specify your own mapping from sample index in the header to SampleInfo object (unique_name, rowIdx) OR
+            //(b) specify the mapping in the callset_mapping_file (JSON) and pass null to addSortedVariantContextIterator()
```

Comment:
The functions have Javadocs  - the lines here repeat information from the function Javadocs